hack/actions/check-changefile-exists.go

```diff
@@ -0,0 +1,136 @@
+//go:build none
+// +build none
+
+// Versioning comment for rerunning jobs.
+// check-changefile-exists.go
+//
+// Checks that the required changelog file exists in the
+// changelogs/unreleased directory.
+//
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v39/github"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+)
+
+func main() {
+
+	log := logrus.StandardLogger()
+	// Forcing colors makes the output nicer to read,
+	// and allows multiline strings to work properly.
+	log.SetFormatter(&logrus.TextFormatter{
+		ForceColors: true,
+	})
+	// We need a GITHUB_TOKEN and PR_NUMBER in the environment.
+	// These are set by the Action config file
+	// in .github/workflows/prbuild.yaml,
+	// under the check-changelog step.
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if !ok {
+		log.Fatal("No GITHUB_TOKEN set, check the Action config.")
+
+	}
+	prEnv, ok := os.LookupEnv("PR_NUMBER")
+	if !ok {
+		log.Fatal("No PR_NUMBER set, check the Action config.")
+	}
+	pr, err := strconv.Atoi(prEnv)
+	if err != nil {
+		log.Fatalf("Couldn't convert PR number, %s", err)
+	}
+
+	// We've got what we need, set up the Github client to get the
+	// labels associated with the PR.
+	ctx := context.Background()
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	prDetails, _, err := client.PullRequests.Get(ctx, "projectcontour", "contour", pr)
+	if err != nil {
+		log.Fatalf("Couldn't get PR details: %s", err)
+	}
+
+	// No labels. This is most likely what people will see at first, so this message
```

Comment:
That's what I was trying to do here - if the initial set of checks fails, then there's no point running the e2es, upgrade tests etc, because they probably won't pass anyway. The initial checks are changefile, linter, codegen, and codespell. These take about a minute to run, so people can get fast feedback on the initial things. Once those are passing, then the longer checks will run.

I definitely tried to make the errors in this script say all the steps needed, but I'll do another pass to check that it's identifying the first thing to fix.

Reply:
Yes absolutely, optimizing on those tests is the BIG win!   

I was only thinking about these very minor steps with the new changelog process, which now do:

1. check if any labels exists in issue -> if not, fail and point out general requirements
2. check if any of the labels can be parsed `release-note/<some string>` -> if not, fail and point out general requirements
3. check if file `./changelogs/unreleased/%d-%s-%s.md` exists -> if not, fail and point the exact name of missing file

in theory, it could do bit more work in advance and pinpoint e.g. "you do not seem to have github label `release-note/<category>` attach to the PR and changelog file `./changelogs/unreleased/%d-%s-%s.md` seems to be missing" at one go.  But this is just minor thing of course!

Reply:
I've added the Action that @stevesloka suggested as a prerequisite for the changefile check, so hopefully the other cases should not come up.

Reply:
Cool! This now of course made label checks in `check-changelog-exists.go` redundant and they could be even removed...

Reply:
I figured I would leave them in as extra precautions.